packages/application-shell/src/components/navbar/navbar.tsx

```diff
@@ -191,10 +191,37 @@ export const ApplicationMenu = (props: ApplicationMenuProps) => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [props.isMenuOpen]); // <-- (re)run this only when the all menu expands
 
+  // Observe submenu to ensure it is not showing on screen when focus is out of it's focus
+  useEffect(() => {
+    const currentRef = submenuRef.current;
+    const observer = new IntersectionObserver((entries) => {
```

Comment:
I'm sorry but I fail to understand this part so I would like to know if you would be up to a sync about it.

Also, bear in mind if we observe an element in the effect, we must unobserve it in the effect cleanup function to avoid memory leaks.

Reply:
Actually this was a leftover from the initial implementation workflow. It was supposed to ensure the open state of the submenu does not persist when we skip it to another element in the dom. 

Removed here: https://github.com/commercetools/merchant-center-application-kit/pull/3482/commits/083c49a0a00a8fb2f187f1db5db125770c866d58

Thank you